src/values/NativeFunctionValue.js

```diff
@@ -83,7 +94,7 @@ export default class NativeFunctionValue extends FunctionValue {
     }
   }
 
-  name: string;
+  name: string | AbstractValue;
```

Comment:
And, whereever `.name` appeared in the past, fill in `.name.throwIfNotConcreteString().value`.